x-pack/plugins/cases/server/services/cases/transform.ts

```diff
@@ -0,0 +1,240 @@
+    connector,
+    references: updatedCase.references,
+    referenceName: connectorIdReferenceName,
+  });
+
+  let externalService: CaseFullExternalService | null | undefined;
+
+  // if external_service is not defined then we don't want to include it in the response since it wasn't passed it as an
+  // attribute to update
+  if (external_service !== undefined) {
+    externalService = transformESExternalService(external_service, updatedCase.references);
+  }
+
+  return {
+    ...updatedCase,
+    attributes: {
+      ...restUpdateAttributes,
+      ...(transformedConnector && { connector: transformedConnector }),
+      // if externalService is null that means we intentionally updated it to null within ES so return that as a valid value
+      ...(externalService !== undefined && { external_service: externalService }),
+    },
+  };
+}
+
+export function transformAttributesToESModel(
+  caseAttributes: CaseAttributes
+): {
+  attributes: ESCaseAttributes;
+  references?: CaseSavedObjectReference[];
+};
+export function transformAttributesToESModel(
+  caseAttributes: Partial<CaseAttributes>
+): {
+  attributes: Partial<ESCaseAttributes>;
+  references?: CaseSavedObjectReference[];
+};
+export function transformAttributesToESModel(
+  caseAttributes: Partial<CaseAttributes>
+): {
+  attributes: Partial<ESCaseAttributes>;
+  references?: CaseSavedObjectReference[];
+} {
+  const { connector, external_service, ...restAttributes } = caseAttributes;
+
+  let transformedAttributes: Partial<ESCaseAttributes> = { ...restAttributes };
+  let pushConnectorId: string | undefined | null;
+
+  if (external_service) {
+    let restExternalService: ExternalServicesWithoutConnectorId | null | undefined;
+    ({ connector_id: pushConnectorId, ...restExternalService } = external_service);
+    transformedAttributes = {
+      ...transformedAttributes,
+      external_service: restExternalService,
+    };
+  } else if (external_service === null) {
+    transformedAttributes = { ...transformedAttributes, external_service: null };
+  }
+
+  if (connector) {
+    transformedAttributes = {
+      ...transformedAttributes,
+      connector: {
+        name: connector.name,
+        type: connector.type,
+        fields: transformFieldsToESModel(connector),
+      },
+    };
+  }
+
+  return {
+    attributes: transformedAttributes,
+    references: buildReferences(connector?.id, pushConnectorId),
+  };
+}
+
+function buildReferences(
+  connectorId?: string,
+  pushConnectorId?: string | null
+): CaseSavedObjectReference[] | undefined {
+  const connectorRef: CaseSavedObjectReference[] = [];
+
+  // this means the reference should be removed
+  if (connectorId === noneConnectorId) {
+    connectorRef.push({ name: connectorIdReferenceName });
+  } else if (connectorId) {
+    connectorRef.push({
+      name: connectorIdReferenceName,
+      ref: { id: connectorId, name: connectorIdReferenceName, type: ACTION_SAVED_OBJECT_TYPE },
+    });
+  }
```

Comment:
nit: 
```suggestion
  connectorRef.push({
      name: connectorIdReferenceName,
      ...(connectorId !== noneConnectorId ? { ref: { id: connectorId, name: connectorIdReferenceName, type: ACTION_SAVED_OBJECT_TYPE }} : {}),
    });
```

Reply:
I refactored this with the introduction of the `ConnectorReferenceHandler`. Let me know what you think.